x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/TransportGetApiKeyAction.java

```diff
@@ -48,6 +48,11 @@ protected void doExecute(Task task, GetApiKeyRequest request, ActionListener<Get
         if (request.ownedByAuthenticatedUser()) {
             assert username == null;
             assert realm == null;
+            if (authentication.getAuthenticatedBy().getType().equals(ApiKeyService.API_KEY_REALM_TYPE)) {
+                throw new IllegalArgumentException(
+                    "failed to retrieve owned API keys for a user authenticated by API key as they cannot own any API keys");
+            }
```

Comment:
Why is this not permitted?

Reply:
An API key can "GET" itself, without any privilege required, and this code here will reject this operation in the case that the owner flag is set. I don't think it should work in this way.

Just to be sure I understood it correctly: we don't wish that the client authenticating with an API key to have any of the `manage_api_key` or `manage_own_api_key` privileges right? If this is correct and this is what this code is trying to achieve, then I suggest we move the code handling this in the privilege/permission space.

Reply:
> An API key can have manage_api_key or manage_own_api_key privilege. If it has manage_own_api_key it can query and manage only itself.

With the code proposed here, I don't think this works, am I right?